q2_types/feature_data/_transformer.py

```diff
@@ -249,12 +250,11 @@ def _fastaformats_to_metadata(ff, constructor=skbio.DNA):
 def _series_to_fasta_format(ff, data, sequence_type="DNA"):
     with ff.open() as f:
         for id_, seq in data.iteritems():
-            if sequence_type == "DNA":
-                sequence = skbio.DNA(seq, metadata={'id': id_})
-                skbio.io.write(sequence, format='fasta', into=f)
+            if sequence_type == "protein":
```

Comment:
Can you leave in the original `else: raise ...` branch? That seems like a useful warning to get when developing future cases. That means we would have `if: protein; elif: DNA; else: raise`.